contentstore/models.py

```diff
@@ -27,3 +36,79 @@ def __unicode__(self):
             rfield(self.minute), rfield(self.hour), rfield(self.day_of_week),
             rfield(self.day_of_month), rfield(self.month_of_year),
         )
+
+
+class MessageSet(models.Model):
+
+    """
+        Details about a set of messages that a recipient can be sent on
+        a particular schedule
+    """
+    short_name = models.CharField(_('Short name'), max_length=20, unique=True)
+    notes = models.TextField(_('Notes'), null=True, blank=True)
+    next_set = models.ForeignKey('self',
+                                 null=True,
+                                 blank=True)
+    default_schedule = models.ForeignKey(Schedule,
+                                         related_name='message_sets',
+                                         null=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __unicode__(self):
+        return "%s" % self.short_name
+
+
+def generate_new_filename(instance, filename):
+    ext = filename.split('.')[-1]  # get file extension
+    return "%s.%s" % (datetime.now().strftime("%Y%m%d%H%M%S%f"), ext)
+
+
+class BinaryContent(models.Model):
+    """
+        File store for reference in messages. Storage method handle by
+        settings file.
+    """
+
+    content = models.FileField(upload_to=generate_new_filename,
+                               max_length=100)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __unicode__(self):
+        return "%s" % (self.content.path.split('/')[-1])
+
+
+class Message(models.Model):
+
+    """
+        Messages that a recipient can be sent
+    """
+    messageset = models.ForeignKey(MessageSet,
+                                   related_name='messages',
```

Comment:
Would `messages` as a related name not be confusing here? Considering we're pointing at the name of a set, not the actual messages?